pkg/scalers/temporal.go

```diff
@@ -0,0 +1,243 @@
+package scalers
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+	sdk "go.temporal.io/sdk/client"
+	sdklog "go.temporal.io/sdk/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+)
+
+var (
+	temporalDefauleQueueTypes = []sdk.TaskQueueType{
+		sdk.TaskQueueTypeActivity,
+		sdk.TaskQueueTypeWorkflow,
+		sdk.TaskQueueTypeNexus,
+	}
+)
+
+type temporalScaler struct {
+	metricType v2.MetricTargetType
+	metadata   *temporalMetadata
+	tcl        sdk.Client
+	logger     logr.Logger
+}
+
+type temporalMetadata struct {
+	Endpoint                  string   `keda:"name=endpoint,                  order=triggerMetadata;resolvedEnv"`
+	Namespace                 string   `keda:"name=namespace,                 order=triggerMetadata;resolvedEnv, default=default"`
+	ActivationTargetQueueSize int64    `keda:"name=activationTargetQueueSize, order=triggerMetadata, default=0"`
+	TargetQueueSize           int64    `keda:"name=targetQueueSize,           order=triggerMetadata, default=5"`
+	QueueName                 string   `keda:"name=queueName,                 order=triggerMetadata;resolvedEnv"`
```

Comment:
Suggest this be named `TaskQueue` rather than `QueueName` so that it lines up with our SDK options.

Reply:
Done